File: converted_source/uwelcome.lfm.py
from common import *

#Useage:
#        import WelcomeWindow
#        window = WelcomeWindow.WelcomeWindow()
#        window.visible = 1

class WelcomeWindow(JFrame):
    def __init__(self, title=' '):
        JFrame.__init__(self, title, windowClosing=self.OnClose)
        self.setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE) # HIDE_ON_CLOSE DO_NOTHING_ON_CLOSE
        self.BuildWindow()
        
    def OnClose(self, event):
        print "Closed"
        
    # a menu is top item on the menu bar and can be pulled down for a list of menu items
    def createMenu(self, name, mnemonic=None, menuBar = None):
        menu = JMenu(name)
        if mnemonic:
            menu.setMnemonic(mnemonic)
        if menuBar:
            menuBar.add(menu)
        return menu
        
    def BuildWindow(self):
        toolkit = java.awt.Toolkit.getDefaultToolkit()
        contentPane = self.getContentPane()
        contentPane.setLayout(None)
        # -- contentPane.setLayout(BoxLayout(contentPane, BoxLayout.Y_AXIS))
        buttonFont = Font("Arial Narrow", Font.PLAIN, 9)
        buttonMargin = Insets(1, 1, 1, 1)
        self.setBounds(375, 189, 247, 250  + 80) # extra for title bar and menu
        #  --------------- UNHANDLED ATTRIBUTE: self.TextHeight = 14
        #  --------------- UNHANDLED ATTRIBUTE: self.BorderIcons = []
        #  --------------- UNHANDLED ATTRIBUTE: self.Scaled = False
        #  --------------- UNHANDLED ATTRIBUTE: self.PixelsPerInch = 96
        #  --------------- UNHANDLED ATTRIBUTE: self.Position = poScreenCenter
        #  --------------- UNHANDLED ATTRIBUTE: self.BorderStyle = bsDialog
        
        self.Label3 = JLabel(' What would you like to do first?', font=buttonFont)
        self.Label3.setBounds(42, 51, 155, 14)
        self.Label1 = JLabel(' Welcome to PlantStudio!', font=buttonFont)
        self.Label1.setBounds(9, 8, 136, 14)
        self.Image1Image = toolkit.createImage("../resources/WelcomeForm_Image1.png")
        self.Image1 = ImagePanel(ImageIcon(self.Image1Image))
        self.Image1.setBounds(3, 1, 241, 45)
        self.hideWelcomeForm = JCheckBox('Don''t show this window next time', font=buttonFont, margin=buttonMargin)
        self.hideWelcomeForm.setMnemonic(KeyEvent.VK_D)
        self.hideWelcomeForm.setBounds(29, 230, 190, 17)
        self.readQuickStartTipList = JButton('Read the Super-Speed Tour', actionPerformed=self.readQuickStartTipListClick, font=buttonFont, margin=buttonMargin)
        self.readQuickStartTipList.setMnemonic(KeyEvent.VK_S)
        self.readQuickStartTipList.setBounds(45, 73, 160, 25)
        self.readTutorial = JButton('Read the Tutorial', actionPerformed=self.readTutorialClick, font=buttonFont, margin=buttonMargin)
        self.readTutorial.setMnemonic(KeyEvent.VK_T)
        self.readTutorial.setBounds(45, 102, 160, 25)
        self.makeNewPlant = JButton('Make a New Plant', actionPerformed=self.makeNewPlantClick, font=buttonFont, margin=buttonMargin)
        self.makeNewPlant.setMnemonic(KeyEvent.VK_N)
        self.makeNewPlant.setBounds(45, 133, 160, 25)
        self.openPlantLibrary = JButton('Open a Plant Library', actionPerformed=self.openPlantLibraryClick, font=buttonFont, margin=buttonMargin)
        self.openPlantLibrary.setMnemonic(KeyEvent.VK_O)
        self.openPlantLibrary.setBounds(45, 162, 160, 25)
        self.startUsingProgram = JButton('Start Using the Program', actionPerformed=self.startUsingProgramClick, font=buttonFont, margin=buttonMargin)
        self.startUsingProgram.setMnemonic(KeyEvent.VK_U)
        self.startUsingProgram.setBounds(45, 191, 160, 25)
        contentPane.add(self.Label3)
        contentPane.add(self.Label1)
        contentPane.add(self.Image1)
        contentPane.add(self.hideWelcomeForm)
        contentPane.add(self.readQuickStartTipList)
        contentPane.add(self.readTutorial)
        contentPane.add(self.makeNewPlant)
        contentPane.add(self.openPlantLibrary)
        contentPane.add(self.startUsingProgram)
        
    def makeNewPlantClick(self, event):
        print "makeNewPlantClick"
        
    def openPlantLibraryClick(self, event):
        print "openPlantLibraryClick"
        
    def readQuickStartTipListClick(self, event):
        print "readQuickStartTipListClick"
        
    def readTutorialClick(self, event):
        print "readTutorialClick"
        
    def startUsingProgramClick(self, event):
        print "startUsingProgramClick"
        
if __name__ == "__main__":
    window = WelcomeWindow()
    window.visible = 1
    window.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE)

File: converted_source/uwelcome.lfm.java

import java.awt.*;
import java.awt.event.*;
import javax.swing.*;
import javax.swing.event.*;
import javax.swing.table.*;
// import common.*;

public class WelcomeWindow extends JFrame {
    private static final long serialVersionUID = 1L;
    JLabel Label3;
    JLabel Label1;
    ImagePanel Image1;
    JCheckBox hideWelcomeForm;
    JButton readQuickStartTipList;
    JButton readTutorial;
    JButton makeNewPlant;
    JButton openPlantLibrary;
    JButton startUsingProgram;
    public JPanel mainContentPane;
    public JPanel delphiPanel;
    
    public WelcomeWindow() {
        super();
        this.setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE); // HIDE_ON_CLOSE DO_NOTHING_ON_CLOSE
        initialize();
    }
    
    private void initialize() {
        this.setSize(new Dimension(705, 325));
        this.setContentPane(getMainContentPane());
        this.setTitle(" ");
    }
    
    private JPanel getMainContentPane() {
        if (mainContentPane == null) {
            mainContentPane = new JPanel();
            mainContentPane.setLayout(new BorderLayout());
            mainContentPane.add(getDelphiPanel(), BorderLayout.CENTER);
        }
        return mainContentPane;
    }
    public void OnClose(ActionEvent event) {
        System.out.println("Closed");
    }
    
    public JPanel getDelphiPanel() {
        if (delphiPanel != null) return delphiPanel;
        Toolkit toolkit = java.awt.Toolkit.getDefaultToolkit();
        delphiPanel = new JPanel();
        delphiPanel.setLayout(null);
        // -- delphiPanel.setLayout(BoxLayout(contentPane, BoxLayout.Y_AXIS));
        // Font buttonFont = new Font("Arial Narrow", Font.PLAIN, 9);
        // Insets buttonMargin = new Insets(1, 1, 1, 1);
        this.setBounds(375, 189, 247, 250  + 80); // extra for title bar and menu
        //  --------------- UNHANDLED ATTRIBUTE: this.TextHeight = 14;
        //  --------------- UNHANDLED ATTRIBUTE: this.BorderIcons = [];
        //  --------------- UNHANDLED ATTRIBUTE: this.Scaled = False;
        //  --------------- UNHANDLED ATTRIBUTE: this.PixelsPerInch = 96;
        //  --------------- UNHANDLED ATTRIBUTE: this.Position = poScreenCenter;
        //  --------------- UNHANDLED ATTRIBUTE: this.BorderStyle = bsDialog;
        
        
        this.Label3 = new JLabel(" What would you like to do first?");
        this.Label3.setBounds(42, 51, 155, 14);
        
        this.Label1 = new JLabel(" Welcome to PlantStudio!");
        this.Label1.setBounds(9, 8, 136, 14);
        
        Image Image1Image = toolkit.createImage("../resources/WelcomeForm_Image1.png");
        this.Image1 = new ImagePanel(new ImageIcon(Image1Image));
        this.Image1.setBounds(3, 1, 241, 45);
        
        this.hideWelcomeForm = new JCheckBox("Don''t show this window next time");
        this.hideWelcomeForm.setMnemonic(KeyEvent.VK_D);
        this.hideWelcomeForm.setBounds(29, 230, 190, 17);
        
        this.readQuickStartTipList = new JButton("Read the Super-Speed Tour");
        this.readQuickStartTipList.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent event) {
                readQuickStartTipListClick(event);
            }
        });
        this.readQuickStartTipList.setMnemonic(KeyEvent.VK_S);
        this.readQuickStartTipList.setBounds(45, 73, 160, 25);
        
        this.readTutorial = new JButton("Read the Tutorial");
        this.readTutorial.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent event) {
                readTutorialClick(event);
            }
        });
        this.readTutorial.setMnemonic(KeyEvent.VK_T);
        this.readTutorial.setBounds(45, 102, 160, 25);
        
        this.makeNewPlant = new JButton("Make a New Plant");
        this.makeNewPlant.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent event) {
                makeNewPlantClick(event);
            }
        });
        this.makeNewPlant.setMnemonic(KeyEvent.VK_N);
        this.makeNewPlant.setBounds(45, 133, 160, 25);
        
        this.openPlantLibrary = new JButton("Open a Plant Library");
        this.openPlantLibrary.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent event) {
                openPlantLibraryClick(event);
            }
        });
        this.openPlantLibrary.setMnemonic(KeyEvent.VK_O);
        this.openPlantLibrary.setBounds(45, 162, 160, 25);
        
        this.startUsingProgram = new JButton("Start Using the Program");
        this.startUsingProgram.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent event) {
                startUsingProgramClick(event);
            }
        });
        this.startUsingProgram.setMnemonic(KeyEvent.VK_U);
        this.startUsingProgram.setBounds(45, 191, 160, 25);
        
        delphiPanel.add(Label3);
        delphiPanel.add(Label1);
        delphiPanel.add(Image1);
        delphiPanel.add(hideWelcomeForm);
        delphiPanel.add(readQuickStartTipList);
        delphiPanel.add(readTutorial);
        delphiPanel.add(makeNewPlant);
        delphiPanel.add(openPlantLibrary);
        delphiPanel.add(startUsingProgram);
        return delphiPanel;
    }
    
        
    public void makeNewPlantClick(ActionEvent event) {
        System.out.println("makeNewPlantClick");
    }
        
    public void openPlantLibraryClick(ActionEvent event) {
        System.out.println("openPlantLibraryClick");
    }
        
    public void readQuickStartTipListClick(ActionEvent event) {
        System.out.println("readQuickStartTipListClick");
    }
        
    public void readTutorialClick(ActionEvent event) {
        System.out.println("readTutorialClick");
    }
        
    public void startUsingProgramClick(ActionEvent event) {
        System.out.println("startUsingProgramClick");
    }

    public static void main(String[] args) {
        SwingUtilities.invokeLater(new Runnable() {
            public void run() {
                WelcomeWindow thisClass = new WelcomeWindow();
                thisClass.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
                thisClass.setVisible(true);
            }
        });
    }

}
